lib/hako/schedulers/ecs.rb

```diff
@@ -872,6 +883,19 @@ def update_service(current_service, task_definition_arn)
           # Keep current desired_count if autoscaling is enabled
           params[:desired_count] = current_service.desired_count
         end
+        # Copy the curreent capacity provider strategy in order to avoid a
+        # perpetual diff when the service specifies no strategy and uses the
+        # cluster's default capacity.
+        # It is not allowed to update the service to use the cluster's default
+        # capacity provider strategy when it is using a non-default capacity
+        # provider strategy.
+        params[:capacity_provider_strategy] ||= current_service.capacity_provider_strategy&.map(&:to_h)
+        if different_capacity_provider_strategy?(params[:capacity_provider_strategy], current_service.capacity_provider_strategy)
+          # Switching from launch type to capacity provider strategy or making
```

Comment:
The exception I've got when trying to update a capacity provider strategy without forcing a new deployment is:

```
Aws::ECS::Errors::InvalidParameterException: When switching from launch type to capacity provider strategy on an existing service, or making a change to a capacity provider strategy on a service that is already using one, you must force a new deployment.
```